kersoene now fuel

LEoJTnTaaa opened the pull request:
kersoene from tfmg now works as fuels

LEoJTnTaaa committed:
kersoene now fuel

Redstonneur1256 commented on src/main/resources/META-INF/mods.toml:
```diff
@@ -45,3 +45,10 @@ mandatory = true
+mandatory=false
+versionRange="[1.0.2c,)"
+ordering="NONE"
+side="BOTH"
```
Why limit the version that players can use for a single fluid? There is no need for TFMG to be declared as an optional dependency.

Redstonneur1256 commented on gradle.properties:
```diff
@@ -1,3 +1,5 @@
 org.gradle.daemon=false
 loom.platform=forge
+github.packages.username=dummy
+github.packages.password=dummy
```
Will break builds, you should define those properties inside your global gradle properties (~/.gradle/gradle.properties); As of right now dependencies depending on it aren't strictly required but will be in the near future so you should replace dummy it with your GitHub username and a token obtained on [your account settings](https://github.com/settings/tokens).

Redstonneur1256 replied:
Will break builds, you should define those properties inside your global gradle properties (~/.gradle/gradle.properties); As of right now dependencies depending on it aren't strictly required but will be in the near future so you should replace dummy it with your GitHub username and a token obtained on [your account settings](https://github.com/settings/tokens).

Redstonneur1256 commented on build.gradle.kts:
```diff
@@ -80,6 +80,10 @@ dependencies {
     modCompileOnly(libs.flywheel.forge.api)
     modRuntimeOnly(libs.flywheel.forge)
 
+    val tfmgCoord = "maven.modrinth:create-tfmg:1.0.2c-1.20.1"
```
Wrong dependency declaration location

Redstonneur1256 commented on build.gradle.kts:
```diff
@@ -80,6 +80,10 @@ dependencies {
 
+    val tfmgCoord = "maven.modrinth:create-tfmg:1.0.2c-1.20.1"
+    modCompileOnly(tfmgCoord)
+    modRuntimeOnly(tfmgCoord)
```
Unnecessary dependency. Optional mods used for testing should be placed under `run/mods-obf/`.

Redstonneur1256 commented:
Also this seems very low effort, there is other fluids that should be compatible such as gasoline, diesel, etc...

LEoJTnTaaa commented:
I know I was unsure on what should and shouldn’t work so I am still working
on it

On Sat, Sep 13, 2025 at 5:53 PM Redstonneur1256 ***@***.***>
wrote:

> *Redstonneur1256* left a comment (Astronauts-of-Create/Northstar-Redux#18)
> <https://github.com/Astronauts-of-Create/Northstar-Redux/pull/18#issuecomment-3288627146>
>
> Also this seems very low effort, there is other fluids that should be
> compatible such as gasoline, diesel, etc...
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Astronauts-of-Create/Northstar-Redux/pull/18#issuecomment-3288627146>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BROXSVSYUHDLLYTM2G4CWRD3SRDXJAVCNFSM6AAAAACGNQ5UNGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTEOBYGYZDOMJUGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

LEoJTnTaaa commented:
I’m getting better tho

On Sun, Sep 14, 2025 at 1:40 AM Leo James ***@***.***> wrote:

> And I am not a great modder and especially hate gradel
>
> On Sun, Sep 14, 2025 at 1:38 AM Leo James ***@***.***> wrote:
>
>> I know I was unsure on what should and shouldn’t work so I am still
>> working on it
>>
>> On Sat, Sep 13, 2025 at 5:53 PM Redstonneur1256 ***@***.***>
>> wrote:
>>
>>> *Redstonneur1256* left a comment
>>> (Astronauts-of-Create/Northstar-Redux#18)
>>> <https://github.com/Astronauts-of-Create/Northstar-Redux/pull/18#issuecomment-3288627146>
>>>
>>> Also this seems very low effort, there is other fluids that should be
>>> compatible such as gasoline, diesel, etc...
>>>
>>> —
>>> Reply to this email directly, view it on GitHub
>>> <https://github.com/Astronauts-of-Create/Northstar-Redux/pull/18#issuecomment-3288627146>,
>>> or unsubscribe
>>> <https://github.com/notifications/unsubscribe-auth/BROXSVSYUHDLLYTM2G4CWRD3SRDXJAVCNFSM6AAAAACGNQ5UNGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZTEOBYGYZDOMJUGY>
>>> .
>>> You are receiving this because you authored the thread.Message ID:
>>> ***@***.***>
>>>
>>

LEoJTnTaaa committed:
All fuels are now compatable and sulfuric acid recipe in tfmg now also compatable

Redstonneur1256 commented on src/main/resources/data/northstar/northstar/fuel_type/tier1.json:
```diff
@@ -2,7 +2,13 @@
+    "tfmg:heavy_oil",
+    "tfmg:diesel",
+    "tfmg:naphtha",
+    "tfmg:creosote"
```
Should crude oil and heavy oil really be used as fuels? Also not all of those have the same recipes and should have their own tiers instead of every fuel being tier 1.

Redstonneur1256 commented on src/main/resources/northstar.mixins.json:
```diff
@@ -1,5 +1,5 @@
 {
-  "required": true,
+  "required": false,
```
Why?

Redstonneur1256 reviewed (changes requested):
Recipes will fail to load when TFMG isn't loaded, they should have a condition on the mod being loaded. See the mekanism biofuel compat recipe for an example.

LEoJTnTaaa committed:
Fix fuel system organization and mixin configuration

- Fixed mixin configuration: set required=true in northstar.mixins.json
- Reorganized fuel system into proper tiers based on real-world performance:
  * Tier 1: High-quality refined fuels (hydrocarbon, gasoline, diesel, kerosene)
  * Tier 0: Basic processed fuels (naphtha, creosote, ethanol) - lower efficiency
  * Crude: Raw petroleum products (crude oil, heavy oil) - very low efficiency
  * Biofuel: Renewable fuel options maintained
- Added WARP.md documentation for development guidance
- Removed dummy credentials from gradle.properties

Addresses issues raised in GitHub PR feedback regarding inappropriate fuel classifications.

LEoJTnTaaa committed:
Delete WARP.md

i used warp becuaase i had troublee with gradle

LEoJTnTaaa committed:
Make TFMG an optional dependency and fix build credential issues

- Remove TFMG from mandatory dependencies in build.gradle.kts and mods.toml
- Set up optional loading from run/mods-obf/ directory
- Fix build issues by making GitHub credentials optional with findProperty
- Fix data type inconsistencies in fuel JSON files (float -> int for combustion_engine_use)
- Tested successful launch with and without TFMG mod present

Redstonneur1256 commented:
Moved to #35
